apps/frontend/src/modules/notifications/slices/notification.slice.ts

```diff
@@ -22,11 +25,17 @@ const { actions, name, reducer } = createSlice({
 			state.dataStatus = DataStatus.PENDING;
 		});
 		builder.addCase(loadAll.fulfilled, (state, action) => {
-			state.notifications = action.payload.items;
+			const { items, totalItems } = action.payload;
+			const { page } = action.meta.arg;
+
+			state.notifications =
+				page === FIRST_PAGE ? items : [...state.notifications, ...items];
+			state.notificationsTotalCount = totalItems;
 			state.dataStatus = DataStatus.FULFILLED;
 		});
 		builder.addCase(loadAll.rejected, (state) => {
 			state.notifications = [];
+			state.notificationsTotalCount = initialState.notificationsTotalCount;
```

Comment:
The idea to use `initialState` for "reseting" is really good, but we don't use it anywhere, so maybe to keep it consistent  let's reset it like
```suggestion
			state.notificationsTotalCount = 0;
```

Reply:
I resolved the conversation, but https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/490#discussion_r1771722196, recommended for this way instead.

Reply:
If we need a specific value it's better to use the initial state or any other constant we have as default. Here we don't have a constant, so initialValue is ok